### ISSUE TITLE
Application Guidelines : CCM, SMap

### ISSUE THREAD

**Issue by SoftwareLiteracy** [SugiharaLab/pyEDM#54]
A series of questions & answers compiled by a pyEDM user:

#### 1. In `CCM` with `libSizes = [10, 70, 10]`, `sample = 100`: What does `sample = 100` mean?

If sample = 100 then at each library size L in [10,20,30,40,50,60,70], L state space vectors are randomly selected and used to predict the target 100 times. The value of correlation shown in the returned DataFrame at each value of L is the average of the simplex prediction of the 100 random state space selections (experiments).

#### 2. The lagged linear correlation between the variables and the target variable may affect the final CCM results. How can we avoid this linear correlation and focus on the nonlinear, state-dependent dynamics that we are interested in?

Geophysical data often have significant serial correlation. If so, EDM results include the linear dependence on short timescales.

To avoid serial correlation if one is not interested in the linear dynamics one can use the `exclusionRadius`. One approach is to assess the lagged mutual information (MI) between the variables and if there is a slow decay in MI set the `exclusionRadius` to the lag at which MI decays (from lag 1) to 1/e or 1/2e.  The linear metric would be use the lag at which the cross correlation goes to 0. If the data are serially correlated this may help resolve the directional coupling.

The sklearn `mutual_info_score` is perhaps better suited for clustering and discrete data sets. The [mutual_info_regression](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_regression.html) is perhaps better suited to time series data.

#### 3. Is there a specific threshold for S-map to distinguish between linear and nonlinear dynamics?

Currently, there are no known general methods, metrics, or thresholds for assessing the state-dependency in S-map. Generally, if predictability is maximum with a global linear regression of the state vectors (theta = 0), the system dynamics may be linear.  If predictability has a local maximum (peak) at values of theta > 0, the dynamics indicate state-dependence and nonlinearity.  However, if the increase in predictability is small and/or the overall prediction skill low, caution is warranted to assess the dynamics as nonlinear. 

#### 4. If both the variables and the target variable exhibit significant seasonal variations, they may share seasonal forcing, leading to spurious CCM results. In other words, if they are both subjected to a common seasonal pattern, CCM analysis may incorrectly suggest a strong causal relationship with the seasonal pattern. How can we address this and distinguish genuine causality?

You can use `SurrogateData` to generate surrogate data and perform significance testing. There is a specific example on page 17 of the [rEDM package documentation](https://cran.r-project.org/web/packages/rEDM/vignettes/rEDM-tutorial.pdf).
